src/include/execution/sql/aggregators.h

```diff
@@ -308,6 +428,71 @@ class TimestampMinAggregate : public MinAggregate<TimestampVal> {};
 /** String min. */
 class StringMinAggregate : public MinAggregate<StringVal> {};
 
+/** Decimal min. */
+class DecimalMinAggregate {
+  static_assert(std::is_base_of_v<Val, DecimalVal>, "Template type must subclass value");
+
+ public:
+  /**
+   * Constructor.
+   */
+  DecimalMinAggregate() : min_(std::numeric_limits<decltype(DecimalVal::val_)>::max()) {
+    uint128_t maxval = 1;
+    maxval = maxval << 127;
+    maxval -= 1;
+    min_.val_.SetValue(maxval);
+    min_.is_null_ = true;
+  }
+
+  /**
+   * This class cannot be copied or moved.
+   */
+  DISALLOW_COPY_AND_MOVE(DecimalMinAggregate);
+
+  /**
+   * Advance the aggregate by the input value @em val.
+   */
+  void Advance(const DecimalVal &val) {
+    if (val.is_null_) {
+      return;
+    }
+    min_.precision_ = val.precision_;
+    min_.is_null_ = false;
+    min_.val_ = std::min(val.val_, min_.val_);
+  }
+
+  /**
+   * Merge a partial min aggregate into this aggregate.
+   */
+  void Merge(const DecimalMinAggregate &that) {
+    if (that.min_.is_null_) {
+      return;
+    }
+    min_.precision_ = that.min_.precision_;
+    min_.is_null_ = false;
+    min_.val_ = std::min(that.min_.val_, min_.val_);
+  }
+
+  /**
+   * Reset the aggregate.
+   */
+  void Reset() {
+    min_.is_null_ = true;
+    uint128_t maxval = 1;
+    maxval = maxval << 127;
```

Comment:
TODO(WAN): I need to look at all the mins and maxes in this file again when I'm not multitasking. Is this correct and is this defined behavior or UB?